source/agora/consensus/SlashPolicy.d

```diff
@@ -279,6 +287,32 @@ public class SlashPolicy
 
         return null;
     }
+
+    /***************************************************************************
+
+        Check if the `missing_validators` data has the information that is
+        slashing a node itself
+
+        Params:
+            missing_validators = list of indices to the validator UTXO set
+                that could be slashed after the consensus process
+
+        Returns:
+            true if the `missing_validators` data has the information that is
+            slashing a node itself.
+
+    ***************************************************************************/
+
+    public bool checkSelfSlashing(in uint[] missing_validators) @safe nothrow
+    {
+        auto enroll_index = this.enroll_man.getIndexOfEnrollment();
+        if (enroll_index != ulong.max &&
+            !missing_validators.find(enroll_index).empty)
+        {
+            return true;
+        }
+        return false;
+    }
```

Comment:
This should probably be in `Ledger` since it's only accessed from there.

Reply:
Moved